src/protobuf-net.Core/Serializers/MapSerializer.cs

```diff
@@ -201,4 +207,54 @@ internal override void Write(ref ProtoWriter.State state, int fieldNumber, WireT
             }
         }
     }
+
+    class IReadOnlyDictionarySerializer<TKey, TValue> : MapSerializer<IReadOnlyDictionary<TKey, TValue>, TKey, TValue>
```

Comment:
Naming... this ... it totally makes sense, but it also gets confusing re interface naming. I wonder if losing the I here would be better.

Reply:
Also, can probably be `sealed`

Reply:
Making it sealed. Naming it as `ReadOnlyDictionarySerializer` would be the most natural to me, but if you would like to keep the fact this is an *interface* then I am thinking of `DictionaryOfIReadOnlyDictionarySerializer`. Do you have a preference or something else on your mind?